waiter/src/waiter/auth/jwt.clj

```diff
@@ -325,6 +325,7 @@
   [{:keys [issuer keys-cache password subject-key supported-algorithms token-type]} request-handler]
   (fn jwt-auth-handler [request]
     (if (and (not (auth/request-authenticated? request))
+             (= "true" (get-in request [:waiter-discovery :service-parameter-template "env" "USE_BEARER_AUTH"]))
```

Comment:
I didn't make this configurable as it is a small change this way. In any case, the tests end up using this environment variable.